stdlib/mexpr/boot-parser.mc

```diff
@@ -308,15 +314,15 @@ let s = "{}" in
 utest lside s with rside s in
 let s = "{a = 5}" in
 utest lside s with rside s in
-let s = "{foo = 123, bar = \"Hello\"}" in
+let s = "{bar = \"Hello\", foo = 123}" in
```

Comment:
The ordering here stems from the order in which the `SID`s were allocated, which in this case depends on which tests were run at which times, right? That means there's a little bit of test interdependence here? Not really sure what to do about it, or if it's fine, at least for now.

Reply:
Right, I think the records are printed in the order of their `SID`s. We could make it so that records are always printed by the order of the actual labels (strings) instead (with a complexity cost, of course).

Reply:
I feel like the cost shouldn't be terribly high given that most records will be pretty small, but I'm also not sure it's worth changing at this point in time. Down the road we might want to do something else anyway, e.g., if we want to use parse and pprint for code formatting, since we probably don't want code formatting to change the printed order of fields.